xc/device.ml

```diff
@@ -1430,6 +1431,116 @@ let can_surprise_remove ~xs (x: device) = Generic.can_surprise_remove ~xs x
 
 module Dm = struct
 
+  module QMP_Event = struct
+    let (pipe_r, pipe_w) = Unix.pipe ()
+    let channel_r = Unix.in_channel_of_descr pipe_r
+    let channel_w = Unix.out_channel_of_descr pipe_w
+
+    let epfd = Epoll.create1 [Epoll.EPOLL_CLOEXEC]
+
+    module Lookup = struct
+      let ftod, dtof = Hashtbl.create 16, Hashtbl.create 16
+
+      let remove domid =
```

Comment:
This is not quite what I had in mind. I meant to have exactly the code below in the Lookup module:
```module Lookup = 
  let ftod, dtof = Hashtbl.create 16, Hashtbl.create 16
  let add fd domid =
    Hashtbl.replace ftod fd domid;
    Hashtbl.replace dtof domid fd
  let remove fd domid =
    Hashtbl.remove ftod fd domid;
    Hashtbl.remove dtof domid fd
  let domid_of fd = try Some (Hashtbl.find ftod fd) with Not_found -> None
  let fd_of domid = try Some (Hashtbl.find dtof domid) with Not_found -> None
end```
and then the QMP_Event.remove() and QMP_Event.add() functions would call the corresponding Lookup.domid_of() and Lookup.fd_of() functions in Lookup.

Reply:
Got it, thanks, updated.